app/screens/home/search/results/results.tsx

```diff
@@ -234,29 +257,55 @@ const SearchResults = ({
         );
     }, [searchValue, selectedTab]);
 
-    let data;
-    if (selectedTab === TabTypes.MESSAGES) {
-        data = orderedPosts;
-    } else {
-        data = orderedFilesForGallery;
-    }
+    const data = selectedTab === TabTypes.MESSAGES ? orderedPosts : orderedFilesForGallery;
+
+    const onActionComplete = useCallback(() => {
+        setSelectedItemNumber(undefined);
+    }, []);
+
+    const renderTabletOptions = useMemo(() => {
+        if (selectedItemNumber === undefined) {
+            return null;
+        }
+
+        const fileInfo = orderedFilesForGallery[selectedItemNumber];
+        return (
+            <AnimatedView
+                style={{
+                    zIndex: tabletZindex,
+                    top: yOffset + tabletTop,
+                }}
+            >
+
+                <FileOptions
+                    fileInfo={fileInfo}
+                    onActionComplete={onActionComplete}
+                    openUp={openUp}
+                />
+            </AnimatedView>
+        );
+    }, [selectedItemNumber, orderedFilesForGallery, onActionComplete, openUp, yOffset]);
```

Comment:
I wonder if this is the best way to solve this. Having to handle at this level makes things a bit convoluted, but maybe doing it a lower level doesn't make it easier.

But if it works, it works :sweat_smile: